pgsync/search_client.py

```diff
@@ -350,7 +350,7 @@ def get_search_client(
 ) -> Union[opensearchpy.OpenSearch, elasticsearch.Elasticsearch]:
     if settings.OPENSEARCH_AWS_HOSTED or settings.ELASTICSEARCH_AWS_HOSTED:
         credentials = boto3.Session().get_credentials()
-        service: str = "es"
+        service: str = "aoss" if settings.OPENSEARCH_AWS_SERVERLESS else "es"
```

Comment:
Do you think we really need this new env var `OPENSEARCH_AWS_SERVERLESS`?
I think this can just be changed to 
`service: str = "aoss" if settings.OPENSEARCH_AWS_HOSTED else "es"`

Reply:
I agree the settings are convoluted, but AWS has two OpenSearch offerings: OpenSearch **Service** and OpenSearch **Serverless**. "Service" needs `es`, while "Serverless" requires `aoss`. I had to add the env var to differentiate them.